components/Dashboard/Settings/UpdatePasswordForm.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react'
+import { useForm } from 'react-hook-form'
+import { toast } from 'react-toastify'
+import { useTranslation } from '../../../config/i18n'
+import FormError from '../../../components/FormError'
+import SettingsForm from '../../../components/Dashboard/Settings/SettingsForm'
+import SettingsFieldset from '../../../components/Dashboard/Settings/SettingsFieldset'
+import Button, { ButtonVariant } from '../../../elements/Button'
+import { useUpdatePasswordMutation } from '../../../generated/graphql'
+import theme from '../../../theme'
+
+type Props = {}
+
+type FormValues = {
+  oldPassword: string
+  newPassword: string
+  confirmNewPassword: string
+}
+
+const UpdatePasswordForm: React.FC<Props> = () => {
+  const { t } = useTranslation('settings')
+
+  const { handleSubmit, register, getValues, errors } = useForm<FormValues>({
+    mode: 'onSubmit',
+    reValidateMode: 'onSubmit',
+  })
+
+  const [updatePassword, { loading }] = useUpdatePasswordMutation({
+    onCompleted: () => {
+      toast.success(t('account.updatePasswordForm.passwordChangeSuccess'))
+    },
+    onError: () => {
+      toast.error(t('account.updatePasswordForm.updateError'))
+    },
+  })
+
+  const handleResetPasswordSubmit = (data: any): void => {
```

Comment:
It would be nice if we could get some type safety on `data`, I think:

```suggestion
  const handleResetPasswordSubmit = (data: FormValues): void => {
```

would work

Reply:
I agree with @Lanny , going to commit the suggestion 🙂